### PR TITLE
[Key Connector] Test that Key Connector URL can be reached before saving

### PR DESCRIPTION
**Pull request by eliykat**
## Type of change
- [ ] Bug fix
- [x] New feature development
- [ ] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [ ] Build/deploy pipeline (DevOps)
- [ ] Other

## Objective

Before saving Key Connector config, ping the Key Connector URL to make sure it's alive.

## Code changes

* **bitwarden_license/src/app/organizations/manage/sso.component.ts** - ping the Key Connector URL on save, return an error toast if it can't be reached.

Depends on https://github.com/bitwarden/jslib/pull/543.

## Testing requirements

Try saving the Key Connector config with valid and invalid Key Connector URLs. If the URL is invalid (i.e. Key Connector cannot be reached), you should get an error toast.

## Before you submit
- [x] I have checked for **linting** errors (`npm run lint`) (required)
- [ ] I have added **unit tests** where it makes sense to do so (encouraged but not required)
- [ ] This change requires a **documentation update** (notify the documentation team)
- [ ] This change has particular **deployment requirements** (notify the DevOps team)

### CODE REVIEW

**Comment by cscharf** [comment]
@eliykat (cc: @Hinton), I think this looks good, not sure if we choose to abandon this or if it's ready to actually merge in or not.

File: bitwarden_license/src/app/organizations/manage/sso.component.ts [hunk 1]
```diff
@@ -108,6 +108,17 @@ export class SsoComponent implements OnInit {
     }
 
     async submit() {
+        if (this.data.get('useKeyConnector').value) {
+            const keyConnectorUrl = this.data.get('keyConnectorUrl').value;
+            try {
+                await this.apiService.getKeyConnectorAlive(keyConnectorUrl);
+            } catch {
+                this.platformUtilsService.showToast('error', null,
+                    'Unable to reach Key Connector. Check the URL and try again.');
```
Comment (eliykat) [inline bitwarden_license/src/app/organizations/manage/sso.component.ts:117]: Needs i18n

**Comment by eliykat** [comment]
I think it's worth merging as an MVP to tick off the item. I've marked it as ready for review in Asana.

**Comment by Hinton** [comment]
@eliykat Do we want to add a button next to the url to allow for manually triggering it as well?

**Comment by eliykat** [comment]
@Hinton I've added a button and in-line validation. Design has been cleared with @danielleflinn as it's not a pattern we otherwise use at the moment.

## Additional code changes

* sso.component.html - change the Test button to an inline/append button. Add success and failure validation messages.
* sso.component.ts - 
  * use `keyConnectorIsValid` to track the validation status of the Key Connector URL. null = needs validation, true = valid, false = invalid.
  * use pristine/dirty to track when the URL needs validation. If it's pristine, we show the error message. As soon as the user interacts with it, the error message disappears until they validate again (either by submitting or using the Test button).
  * I have a WIP trying to do this with a custom async validator, which should be possible, but I couldn't get it working in time.

## Screenshots

Screen Shot 2021-11-19 at 1 37 47 pm

Screen Shot 2021-11-19 at 1 38 01 pm

Screen Shot 2021-11-19 at 1 38 35 pm

**Review by Hinton — APPROVED** [review]

**Pull request merged**